napari/_qt/dialogs/qt_notification.py

```diff
@@ -301,17 +295,46 @@ def sizeHint(self):
         )
 
     @classmethod
-    def from_exception(cls, exception: BaseException) -> 'NapariNotification':
-        """Create a NapariNotifcation dialog from an exception object."""
-        # TODO: this method could be used to recognize various exception
-        # subclasses and populate the dialog accordingly.
-        extra_msg = (
-            "\nYou can start napari with NAPARI_CATCH_ERRORS=0 or "
-            "NAPARI_EXIT_ON_ERROR=1 to find more about this error"
+    def from_notification(cls, notification):
+        actions = notification.actions
+
+        if not actions and isinstance(notification, ErrorNotification):
+            # TODO: this should be moved
+
+            exc = notification.exception
+
+            def show_tb(parent):
+                tbdialog = QDialog(parent=parent.parent())
+                tbdialog.setModal(True)
+                tbdialog.setLayout(QVBoxLayout())
+                try:
+                    format_exc_info = get_tb_formatter()
+                except Exception:
+                    import traceback
+                    traceback.print_tb()
+                    raise
+                info = (exc.__class__, exc, exc.__traceback__)
+                text = QTextEdit()
+                text.setHtml(format_exc_info(info, as_html=True))
+                text.setReadOnly(True)
+                print()
+                layout = tbdialog.layout()
+                print()
+                layout.addWidget(text)
+                tbdialog.show()
+
+            actions = [('traceback', show_tb)]
+
+        dialog = cls(
+            message=notification.message,
+            severity=notification.severity,
+            source=notification.source,
+            actions=actions,
         )
 
-        msg = getattr(exception, 'message', str(exception)) + extra_msg
-        severity = getattr(exception, 'severity', 'WARNING')
-        source = None
-        actions = getattr(exception, 'actions', ())
-        return cls(msg, severity, source, actions)
+        return dialog
+
+    @staticmethod
+    def show_notification(notification):
```

Comment:
I'm thinking that static and class methods might be problematic as we don't know if the caller is going to keep a reference.

Reply:
Agreed!